code/modules/clothing/head/jobs.dm

```diff
@@ -146,12 +149,55 @@
 	else
 		to_chat(user, "You just took a candy corn! You should wait a couple minutes, lest you burn through your stash.")
 
+/obj/item/clothing/head/fedora/det_hat/ui_action_click(mob/user, action)
+	if(user.mind?.assigned_role != JOB_NAME_DETECTIVE)
+		to_chat(user, "<span class='warning>Only a true detective can use the noir ability.</span>")
+		return
+	var/turf/T = get_turf(user)
+	if (istype(T.loc, /area/security/detectives_office))
+		for(var/mob/living/carbon/human/H in T.loc)
+			ADD_TRAIT(H, TRAIT_NOIR, TRAIT_GENERIC)
+			H.add_client_colour(/datum/client_colour/monochrome)
+			if(H.mind?.assigned_role == JOB_NAME_DETECTIVE)
+				to_chat(H, span_notice("The shadows overtake your office. They are in your realm now."))
+			else
+				to_chat(H, span_userdanger("The shadows overtake the detective's office. An omnious feeling takes over you"))
+	else
```

Comment:
early return this instead of an if else conditional, its good practice to reduce nesting wherever possible